### ISSUE TITLE
Rainier - Add DPS310 ambient & altitude sensor configurations

### ISSUE THREAD

**Issue by msbarth** [ibm-openbmc/dev#2652]

**Comment by msbarth** [comment]
Assigning to Eddie. After a conversation with him, he's working on getting the DPS310 data onto dbus which covers this issue and I didnt find another issue referencing the DPS310 work.

**Comment by mzipse** [comment]
@BMC-Bruce , are you covering this with Eddie unavailable?  Reassigning to Bruce.

**Comment by mzipse** [comment]
You might want to confirm with Matt as to what is expected.

**Comment by BMC-Bruce** [comment]
> @BMC-Bruce , are you covering this with Eddie unavailable? Reassigning to Bruce.
Yes, I am.

**Comment by BMC-Bruce** [comment]
> You might want to confirm with Matt as to what is expected.

I will connect with Matt as to expectations.  Thanks!

**Comment by msbarth** [comment]
@BMC-Bruce 
The expectations are to get both the ambient and pressure(altitude) sensors values from the DPS310 into userspace and on dbus. I believe Eddie was working on adding support to dbus-sensors for the iio subsystem to support getting these on dbus using dbus-sensors.

**Comment by BMC-Bruce** [comment]
@rfrandse just merged this PR meta-ibm: Adds virtual Altitude sensor #1164 into 1020

To read the Altitude reading use:
busctl introspect xyz.openbmc_project.VirtualSensor /xyz/openbmc_project/sensors/altitude/Altitude

To read the DPS310 Pressure sensor use:
busctl introspect --no-pager xyz.openbmc_project.IIOSensor /xyz/openbmc_project/sensors/pressure/Station_Pressure

To read the DPS310 Temperature sensor use:
busctl introspect --no-pager xyz.openbmc_project.IIOSensor /xyz/openbmc_project/sensors/temperature/Ambient_1_Temp

**Comment by BMC-Bruce** [comment]
Just a status update:
@msbarth So I believe @eddiejames is working on the Everest Device Tree for the DPS310 sensors.
On Everest I presently see no temperature sensors what so ever and certainly not the DPS310.
entity-manager will also need some work for Everest support, possibly even the Virtual Sensors if sensor names/identifiers change.

**Comment by BMC-Bruce** [comment]
Rainier / Everest Si7021-A20 ambient on dbus pending on 2 PRs
entity-manager: Support for DPS310 and SI7020 Rainier #5
and
dbus-sensors: Adds support for DPS310 Sensors #5

**Comment by BMC-Bruce** [comment]
The Altitude Virtual Sensor is handle by the Gerrit Merged Review.
44526: meta-ibm: Adds virtual Altitude sensor
https://gerrit.openbmc-project.xyz/c/openbmc/openbmc/+/44526

**Comment by BMC-Bruce** [comment]
New PRs, old ones closed.

entity-manager
1020 dps310 and SI7020 #6
https://github.ibm.com/openbmc/entity-manager/pull/6

dbus-sensors: Support DPS310 and SI7020 #6
https://github.ibm.com/openbmc/dbus-sensors/pull/6

**Comment by BMC-Bruce** [comment]
To read Ambient Temperatures, Station Pressure, and Altitude

```
busctl introspect --no-pager xyz.openbmc_project.HwmonTempSensor /xyz/openbmc_project/sensors/temperature/Ambient_0_Temp
busctl introspect --no-pager xyz.openbmc_project.IIOSensor /xyz/openbmc_project/sensors/temperature/Ambient_1_Temp
busctl introspect --no-pager xyz.openbmc_project.IIOSensor /xyz/openbmc_project/sensors/temperature/Ambient_2_Temp
busctl introspect --no-pager xyz.openbmc_project.IIOSensor /xyz/openbmc_project/sensors/pressure/Station_Pressure

busctl introspect xyz.openbmc_project.VirtualSensor /xyz/openbmc_project/sensors/temperature/Ambient_Virtual_Temp
busctl introspect xyz.openbmc_project.VirtualSensor /xyz/openbmc_project/sensors/altitude/Altitude

```

**Comment by BMC-Bruce** [comment]
44290: entity-manager: Support for DPS310 and SI7020 Rainier
       https://gerrit.openbmc-project.xyz/c/openbmc/entity-manager/+/44290
44670: entity-manager: Support DPS310 and SI7020 Everest
       https://gerrit.openbmc-project.xyz/c/openbmc/entity-manager/+/44670
43410: dbus-sensors: Support DPS310 and SI7020
       https://gerrit.openbmc-project.xyz/c/openbmc/dbus-sensors/+/43410

In reality 44290 is a prerequisite for 44670.
And Ed has In reality made 43410 a prerequisite for both 44290 and 44670.

**Comment by BMC-Bruce** [comment]
Closing this for eBMC as for GA1 we will be using `dbus-sensors` IIOSensor,
upstream will not have IIOSensor and it will be merged into HwmonTempSensor.
But that will happen latter.
